plugins/applications/src/lib.rs

```diff
@@ -105,13 +105,33 @@ pub fn get_matches(input: RString, state: &State) -> RVec<Match> {
         .entries
         .iter()
         .filter_map(|(entry, id)| {
-            let score: i64 = matcher.fuzzy_match(&entry.name, &input).unwrap_or(0)
-                + matcher.fuzzy_match(&entry.exec, &input).unwrap_or(0)
-                + entry
-                    .keywords
-                    .iter()
-                    .map(|keyword| matcher.fuzzy_match(keyword, &input).unwrap_or(0))
-                    .sum::<i64>();
+            // For desktop actions we still want the match
+            // on the applications name, so we match the description.
+            // We substract the offset so that action entries are prioritized
+            // by order of definition if they match for the same application.
+            let name_score = match &entry.desc {
+                None => matcher.fuzzy_match(&entry.name, &input).unwrap_or(0),
+                Some(val) => matcher.fuzzy_match(&val, &input).unwrap_or(0) - entry.offset,
+            };
+
+            // We need this to avoid non-action entries get double score.
+            let exec_score = match &entry.desc {
+                None => 0,
+                Some(_) => matcher.fuzzy_match(&entry.name, &input).unwrap_or(0),
+            };
+
+            let keyword_score = entry
+                .keywords
+                .iter()
+                .map(|keyword| matcher.fuzzy_match(keyword, &input).unwrap_or(0))
+                .sum::<i64>();
+
+            let mut score = name_score * 50 + exec_score * 25 + keyword_score;
+
+            // prioritize actions
+            if entry.desc.is_some() {
```

Comment:
I'm not that sure about prioritizing desktop actions over the main desktop entries, but I guess it is a matter of testing if it is a good experience in the end.

Reply:
Yeah, i agree. Maybe if there's bad user feedback regarding this, we can simply make this optional.

Reply:
Sure, let's see how it goes.